modules/placekey/redshift/lib/__init__.py

```diff
@@ -0,0 +1,356 @@
+
+def placekey_to_h3(placekey):
+    """
+    Convert a Placekey string into an H3 hexadecimal string.
+    :param placekey : Placekey index (string)
+    :return: H3 (string)
+    """
+    return int_to_h3_string(placekey_to_h3_integer(placekey))
+
+
+# Auxiliary functions
+
+def string_to_h3(h3_index):
+    result = 0
+    if type(h3_index) != str: 
+        return result
+
+    try: 
+        result = int(h3_index, 16)
+    except:
+        result = 0
+
+    return result
+
+def encode_h3_int(h3_integer):
+    short_h3_integer = shorten_h3_integer(h3_integer)
+    encoded_short_h3 = encode_short_int(short_h3_integer)
+
+    clean_encoded_short_h3 = clean_string(encoded_short_h3)
+    if len(clean_encoded_short_h3) <= CODE_LENGTH:
+        clean_encoded_short_h3 = str.rjust(clean_encoded_short_h3, CODE_LENGTH, PADDING_CHAR)
+
+    return '@' + '-'.join(clean_encoded_short_h3[i:i + TUPLE_LENGTH]
+                          for i in range(0, len(clean_encoded_short_h3), TUPLE_LENGTH))
+
+def shorten_h3_integer(h3_integer):
+    # Cuts off the 12 left-most bits that don't code location
+    out = (h3_integer + BASE_CELL_SHIFT) % (2 ** 52)
+    # Cuts off the rightmost bits corresponding to resolutions greater than the base resolution
+    out = out >> (3 * (15 - BASE_RESOLUTION))
+    return out
+
+def encode_short_int(x):
+    if x == 0:
+        return ALPHABET[0]
+    else:
+        res = ''
+        while x > 0:
+            remainder = x % ALPHABET_LENGTH
+            res = ALPHABET[remainder] + res
+            x = x // ALPHABET_LENGTH
+        return res
+
+def clean_string(s):
+    # Replacement should be in order
+    for k, v in REPLACEMENT_MAP:
+        if k in s:
+            s = s.replace(k, v)
+    return s
+
+
+def parse_plakey(placekey):
+    if '@' in placekey:
+        what, where = placekey.split('@')
+    else:
+        what, where = None, placekey
+
+    return what, where
+
+def placekey_to_h3_integer(placekey):
+    _, where_part = parse_plakey(placekey)
+    code = strip_encoding(where_part)
+    dirty_encoding = dirty_string(code)
+    short_h3_integer = decode_string(dirty_encoding)
+    return unshorten_h3_integer(short_h3_integer)
+
+def strip_encoding(s):
+    s = s.replace('@', '').replace('-', '').replace(PADDING_CHAR, '')
+    return s
+
+def dirty_string(s):
+    # Replacement should be in (reversed) order
+    for k, v in REPLACEMENT_MAP[::-1]:
+        if v in s:
+            s = s.replace(v, k)
+    return s
+
+def decode_string(s):
+    val = 0
+    for i in range(len(s)):
+        val += (ALPHABET_LENGTH ** i) * ALPHABET.index(s[-1 - i])
+    return val
+
+def unshorten_h3_integer(short_h3_integer):
+    unshifted_int = short_h3_integer << (3 * (15 - BASE_RESOLUTION))
+    rebuilt_int = ((HEADER_INT + UNUSED_RESOLUTION_FILLER) - BASE_CELL_SHIFT) + unshifted_int
+
+    return rebuilt_int
+
+def int_to_h3_string(h3_integer):
+    return str(np.base_repr(h3_integer, 16)).lower()
```

Comment:
if you do the import numpy, inside this function and remove it from the top, very likely it won't be needed in the python3_requirements.txt. At least that worked for me .

Reply:
Finally, I have removed any import from numpy since it seems that Python 2 manages well 64 bit integer natively.